File: src/main/java/org/pyjjs/scheduler/core/api/impl/actors/task/behaviours/TaskInitBehaviour.java
package org.pyjjs.scheduler.core.api.impl.actors.task.behaviours;

import org.pyjjs.scheduler.core.api.impl.actors.common.behaviours.Behaviour;
import org.pyjjs.scheduler.core.api.impl.actors.task.TaskActorState;
import org.pyjjs.scheduler.core.api.impl.actors.task.supervisor.messages.TaskInitMessage;

public class TaskInitBehaviour extends Behaviour<TaskActorState, TaskInitMessage> {

    @Override
    protected void perform(TaskInitMessage message) {
        TaskActorState taskActorState = getActorState();
        taskActorState.setSource(message.getSource());
        taskActorState.setInitialized(true);
        saveActorState(taskActorState);
    }

    @Override
    protected Class<TaskInitMessage> processMessage() {
        return TaskInitMessage.class;
    }
}

File: src/main/kotlin/org/pyjjs/scheduler/core/api/impl/actors/task/behaviours/TaskInitBehaviour.kt
package org.pyjjs.scheduler.core.api.impl.actors.task.behaviours

import org.pyjjs.scheduler.core.api.impl.actors.common.messages.TaskInitMessage
import kotlin.reflect.KClass

class TaskInitBehaviour : TaskBehaviour<TaskInitMessage>() {

    override fun perform(message: TaskInitMessage) {
        val taskActorState = actorState
        taskActorState.source = message.source
        taskActorState.isInitialized = true
        saveActorState(taskActorState)
    }

    override fun processMessage(): KClass<TaskInitMessage> {
        return TaskInitMessage::class
    }
}
